File: php/src/Eratosthenes/FixedSieve.php
<?php

namespace App\Eratosthenes;

use SplFixedArray;

class FixedSieve
{
    use SieveTrait;

    protected SplFixedArray $isPrime;

    public function __construct(int $size)
    {
        $this->size    = $size;
        $this->isPrime = new SplFixedArray($size + 1);
    }
}

File: c#/Benchmark/Logic/Eratosthenes/FixedSieve.cs
namespace Benchmark.Logic.Eratosthenes;

public class FixedSieve : AbstractSieve
{
    public FixedSieve(int size)
    {
        this.Size    = size;
        this.IsPrime = new bool[this.Size + 1];
    }

    protected override IList<bool> IsPrime { get; }
}
